### PR TITLE
allow embed in send_file

### PR DESCRIPTION
**Pull request by brakhane**
embed can contain an image, and that image can be a message attachment.
Update the code to allow this use case.

Also, the pool of characters for the boundary string was weird, so fix
this as well.

### CODE REVIEW

**Comment by Fuyukai** [comment]
The boundary string was keymash + 69420 on the end, not that weird tbh.

**Comment by Fuyukai** [comment]
Also I'm honestly surprised this was that simple, I've been putting this off for well over a year because embeds and attachments never played well.

File: curious/dataclasses/embed.py [hunk 1]
```diff
@@ -151,8 +151,8 @@ def set_image(self, *, image_url: str) -> 'Embed':
         """
         self.image = attrdict()
 
-        if not image_url.startswith("http") or image_url.startswith("attachment://"):
-            raise ValueError("Image URLs must start with http[s]")
+        if not (image_url.startswith("http") or image_url.startswith("attachment://")):
```
Comment (Fuyukai) [inline curious/dataclasses/embed.py:154]: What's with the extra brackets?

**Comment by brakhane** [comment]
> The boundary string was keymash + 69420 on the end, not that weird tbh.

The that wasn't the boundary string, but the pool from which 30 elements were taken to create the boundary string

File: curious/dataclasses/embed.py (hunk 1 above)
Comment (brakhane) [reply curious/dataclasses/embed.py:154]: `not False or True` is `True`, we want `not (False or True)`, which is `False`

Comment (Fuyukai) [reply curious/dataclasses/embed.py:154]: Oh okay I don't know how to program

**Comment by brakhane** [comment]
> Also I'm honestly surprised this was that simple,

It took me around an hour or so to get it working because the Discord API docs were so confusing, and also wrong.

Even discord.py didn't implement it, so curious is now the only Python Discord API that does

**Pull request merged**